tikv/tikv
# Add query processor metrics

Signed-off-by: Sunny Bains <sunny.bains@pingcap.com>



### What is changed and how it works?
Issue Number: Close #13027 

What's Changed:

Add metrics to track coprocessor query executor memory usage.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)

  Start cluster using tiup
  Add the Prometheus query from tikv/metrics/grafana/tikv_details.json to the coprocessor details dashboard
  Connect to the cluster using the MySQL client  and run the SQL.

   create table t (c1 int, c2 int, c3 int);
   insert into t values(-1, null, null), (-10, null, -5), (null, -1, -1), (-10, -10, 0), (null, null, 2), (null, null, 2), (-10, -10, 3), (null, 1, 4);
  insert into t select * from t;
  insert into t select * from t;
  insert into t select * from t;
  insert into t select * from t;
  insert into t select * from t;
  insert into t select * from t;
  insert into t select * from t;
  insert into t select * from t;
  insert into t select * from t;
  insert into t select * from t;
  insert into t select * from t;
  insert into t select * from t;
  insert into t select * from t;
  insert into t select * from t;
  insert into t select * from t;
  insert into t select * from t;
  select * from t group by c2,c1,c3;
  select * from t order by c2 limit 10;

  You should be able to see the memory usage go up for the relevant query processor and then go down to zero once the query completes.

Side effects

None

### Release note 

```release-note
Support tracking of  Coprocessor Query Executor memory, this should help in diagnosing memory consumption.
```
